File: src/static/cloud/src/routes/local.js
/*
 * Routes for locally stored historic data endpoints
 * @param Api <Express>
 * @param LocalController <Controller>
 * @param basePath <string>
 */
export default (Api, LocalController, basePath) => {
  /*
   * Create base paths for various endpoints
   */
  const base = `${process.env.DB_API}/${basePath}`

  /*
   * Get Exchanges
   * @param name <[<string>]> [optional]
   * @param fields <obj name, label, etc...> [optional]
   */
  Api.get(`${base}/exchanges`, LocalController.getExchanges)

  /*
   * Get Products
   * @param pair <[<string>]> [optional]
   * @param fields <obj pair, base, quote, etc...> [optional]
   */
  Api.get(`${base}/products`, LocalController.getProducts)

  /*
   * Get Klines
   * @param pair <[<string>]>
   * @param fields <obj name, label, etc...> [optional]
   * @param start <datetime> [optional]
   * @param end <datetime> [optional]
   * @param granularity <int> [optional]
   * @param limit <int> [optional]
   */
  Api.get(`${base}/klines`, LocalController.getKlines)

  /*
   * Get Tickers
   * @param pair <[<string>]>
   * @param fields <obj name, label, etc...> [optional]
   * @param start <datetime> [optional]
   * @param end <datetime> [optional]
   * @param granularity <int> [optional]
   * @param limit <int> [optional]
   */
  Api.get(`${base}/tickers`, LocalController.getTickers)

  /*
   * Get Trades
   * @query pair <[<string>]>
   * @param fields <obj name, label, etc...> [optional]
   * @param start <datetime> [optional]
   * @param end <datetime> [optional]
   * @param granularity <int> [optional]
   * @param limit <int> [optional]
   */
  Api.get(`${base}/trades`, LocalController.getTrades)
}

File: src/static/bootstrap/cryptodock/local.py
import requests

class Local :

    def __init__(self, uri) :
        self.uri = uri

    def get(self, endpoint, params = {}) :
        response = requests.get(self.uri + '/local' + endpoint, params)
        return response.json() if response.status_code > 199 and response.status_code < 300 else False

    def get_exchanges(self, exchanges) :
        return self.get("/exchanges", {
            'exchanges': exchanges,
        })

    def get_products(self, pairs, exchanges, fields) :
        return self.get("/products", {
            'pairs': pairs,
            'exchanges': exchanges,
            'fields': fields
        })

    def get_tickers(self, pairs, exchanges, fields, start = False, end = False, granularity = 3600, limit = -1) :
        return self.get("/tickers", {
            'pairs': pairs,
            'exchanges': exchanges,
            'fields': fields,
            'start': start,
            'end': end,
            'granularity': granularity,
            'limit': limit
        })


    def get_trades(self, pairs, exchanges, fields, start = False, end = False, granularity = 3600, limit = -1) :
        return self.get("/trades", {
            'pairs': pairs,
            'exchanges': exchanges,
            'fields': fields,
            'start': start,
            'end': end,
            'granularity': granularity,
            'limit': limit
        })

    def get_klines(self, pairs, exchanges, fields, start = False, end = False, granularity = 3600, limit = -1) :
        return self.get("/klines", {
            'pairs': pairs,
            'exchanges': exchanges,
            'fields': fields,
            'start': start,
            'end': end,
            'granularity': granularity,
            'limit': limit
        })
